vault/resource_ad_secret_backend.go

```diff
@@ -0,0 +1,654 @@
+			Description: `CA certificate to use when verifying LDAP server certificate, must be x509 PEM encoded.`,
+		},
+		"client_tls_cert": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Sensitive:   true,
+			Description: `Client certificate to provide to the LDAP server, must be x509 PEM encoded.`,
+		},
+		"client_tls_key": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Sensitive:   true,
+			Description: `Client certificate key to provide to the LDAP server, must be x509 PEM encoded.`,
+		},
+		"default_lease_ttl_seconds": {
+			Type:        schema.TypeInt,
+			Optional:    true,
+			Computed:    true,
+			Description: "Default lease duration for secrets in seconds",
+		},
+		"deny_null_bind": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Description: `Denies an unauthenticated LDAP bind request if the user's password is empty; defaults to true`,
+		},
+		"description": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "Human-friendly description of the mount for the backend.",
+		},
+		"discoverdn": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Description: `Use anonymous bind to discover the bind DN of a user.`,
+		},
+		"formatter": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Computed:    true,
+			Deprecated:  `Formatter is deprecated and password_policy should be used.`,
+			Description: `Text to insert the password into, ex. "customPrefix{{PASSWORD}}customSuffix".`,
+		},
+		"groupattr": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: `LDAP attribute to follow on objects returned by <groupfilter> in order to enumerate user group membership. Examples: "cn" or "memberOf", etc. Default: cn`,
+		},
+		"groupdn": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: `LDAP search base to use for group membership search (eg: ou=Groups,dc=example,dc=org)`,
+		},
+		"groupfilter": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: `Go template for querying group membership of user. The template can access the following context variables: UserDN, Username Example: (&(objectClass=group)(member:1.2.840.113556.1.4.1941:={{.UserDN}})) Default: (|(memberUid={{.Username}})(member={{.UserDN}})(uniqueMember={{.UserDN}}))`,
+		},
+		"insecure_tls": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Description: `Skip LDAP server SSL Certificate verification - insecure and not recommended for production use.`,
+		},
+		"last_rotation_tolerance": {
+			Type:        schema.TypeInt,
+			Optional:    true,
+			Computed:    true,
+			Description: `The number of seconds after a Vault rotation where, if Active Directory shows a later rotation, it should be considered out-of-band.`,
+		},
+		"length": {
+			Type:        schema.TypeInt,
+			Optional:    true,
+			Computed:    true,
+			Deprecated:  `Length is deprecated and password_policy should be used.`,
```

Comment:
Same suggestion here:
```suggestion
			Deprecated:  `Length is deprecated and password_policy should be used with Vault >= 1.5.`,
```